pylint/testutils/_primer/primer_run_command.py

```diff
@@ -8,63 +8,52 @@
 import sys
 import warnings
 from io import StringIO
-from itertools import chain
 
 from pylint.lint import Run
 from pylint.message import Message
 from pylint.reporters import JSONReporter
+from pylint.reporters.json_reporter import OldJsonExport
 from pylint.testutils._primer.package_to_lint import PackageToLint
-from pylint.testutils._primer.primer_command import PackageMessages, PrimerCommand
+from pylint.testutils._primer.primer_command import PrimerCommand
 
 GITHUB_CRASH_TEMPLATE_LOCATION = "/home/runner/.cache"
 CRASH_TEMPLATE_INTRO = "There is a pre-filled template"
 
 
 class RunCommand(PrimerCommand):
     def run(self) -> None:
-        packages: PackageMessages = {}
-
+        packages: dict[str, list[OldJsonExport]] = {}
         for package, data in self.packages.items():
-            packages[package] = self._lint_package(data)
-            print(f"Successfully primed {package}.")
+            packages[package] = self._lint_package(package, data)
+        path = (
+            self.primer_directory
+            / f"output_{'.'.join(str(i) for i in sys.version_info[:3])}_{self.config.type}.txt"
+        )
+        print(f"Writing result in {path}")
+        with open(path, "w", encoding="utf-8") as f:
+            json.dump(packages, f)
 
+    def _get_astroid_errors(
+        self, messages: list[OldJsonExport]
+    ) -> tuple[list[Message], list[Message]]:
         astroid_errors = []
         other_fatal_msgs = []
-        for msg in chain.from_iterable(packages.values()):
-            if msg.category == "fatal":
-                if GITHUB_CRASH_TEMPLATE_LOCATION in msg.msg:
+        for raw_message in messages:
+            message = JSONReporter.deserialize(raw_message)
```

Comment:
We also de-serialize messages only if we need to when there are fatal error messages. At this point we could not deserialize, at all but heh.